---
title: "RNAseq Results"
author: "Heath O'Brien"
output:
  tufte::tufte_html: default
  #tufte::tufte_handout: default
---

```{r setup, include=FALSE}
library(tidyverse)
library(devtools)
library(stringr)
library(tufte)
LibraryInfo=read_delim("../conditions.txt", "\t")

plotExpression <- function(row, LibraryInfo, title){
    dplyr::slice(row, 1) %>%
    select(one_of(paste('norm', LibraryInfo$SampleID, sep='.'))) %>%
    gather("SampleID", "Count") %>%
    mutate(SampleID = str_extract(SampleID, '[^.]+$')) %>%
    full_join(LibraryInfo) %>%
    mutate(condition=factor(condition, levels= c('Uninfected', 'WT', 'hog1'))) %>%
    ggplot(aes(y=Count, x=condition, colour=condition)) +
    geom_jitter(height = 0, width=.1) +
    ggtitle(title)
}


```

## The vast majority of reads map to Manduca contigs
```{r coverage, echo=FALSE, message=FALSE, warning=FALSE}
folder="../Mappings"
conditions <- read_tsv("../conditions.txt")
cov <- data.frame()
for (file in dir(folder, '*cov.txt')) {
  cov <- read_delim(paste(folder, file, sep='/'), "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE) %>%
    mutate(sample=head(strsplit(file, '\\.')[[1]], 1)) %>%
    rbind(cov)
}

cov <- mutate(cov, Organism=ifelse(str_detect(X1, '^gi'), 'Manduca', 'Candida'))

for (file in dir(folder, '*stats.txt')) {
  temp <- read_csv(paste(folder, file, sep='/'))
  colnames(temp)=c("x1")
  cov <-filter(temp, str_detect(x1, "Unmapped reads")) %>% separate(x1, into=c("Organism", "X2", "X3")) %>% 
    mutate(X1=NA, X4=NA, sample=head(strsplit(file, '\\.')[[1]], 1)) %>%
    rbind(cov) %>%
    mutate(X2=as.integer(X2), X3=as.integer(X3), X4=as.integer(X4))
}

group_by(cov, Organism, sample) %>% 
  summarise(total_reads=sum(as.integer(X3))) %>% 
  full_join(conditions, by=c("sample" = "SampleID")) %>% 
  ggplot(aes(x=sample, y=total_reads, fill=condition)) +
    geom_bar(stat='identity') +
    facet_grid(Organism~.) 

group_by(cov, Organism, sample) %>% 
  summarise(total_reads=sum(as.integer(X3))) %>% 
  full_join(conditions, by=c("sample" = "SampleID")) %>% 
  ggplot(aes(x=sample, y=total_reads, fill=condition)) +
    geom_bar(stat='identity') +
    facet_grid(Organism~.) +
  scale_y_log10()

```

## Summary of all DE genes (FDR<0.1)
```{r all_DE, echo=FALSE, message=FALSE}
DE <- mutate(read_delim("../Results/tables/Uninfectedvshog1.down.txt", "\t"), Comparison="Uninfectedvshog1", bias='hog1' ) %>%
  bind_rows(mutate(read_delim("../Results/tables/Uninfectedvshog1.up.txt", "\t"), Comparison="Uninfectedvshog1", bias='Uninfected' )) %>%
  bind_rows(mutate(read_delim("../Results/tables/WTvshog1.down.txt", "\t"), Comparison="WTvshog1", bias='hog1' )) %>%
  bind_rows(mutate(read_delim("../Results/tables/WTvshog1.up.txt", "\t"), Comparison="WTvshog1", bias='WT' )) %>%
  bind_rows(mutate(read_delim("../Results/tables/WTvsUninfected.down.txt", "\t"), Comparison="WTvsUninfected", bias='Uninfected' )) %>%
  bind_rows(mutate(read_delim("../Results/tables/WTvsUninfected.up.txt", "\t"), Comparison="WTvsUninfected", bias='WT' ))

DE <- DE %>% mutate(up = ifelse(log2FoldChange>0, str_remove(Comparison, "vs.*"), str_remove(Comparison, ".*vs")), down = ifelse(log2FoldChange<0, str_remove(Comparison, "vs.*"), str_remove(Comparison, ".*vs")))

knitr::kable(
  group_by(DE, Comparison, bias) %>% summarise(n=n())
)
group_by(DE, Id, bias) %>% 
  summarise(n=n()) %>% 
  filter(n>1) %>% 
  select(Id) %>% 
  left_join(DE) %>% 
  select(Id, Comparison, bias, log2FoldChange, padj, everything()) %>% 
  write_tsv("../Results/tables/TransitiveDiffs.txt")
```

## Summary of transitive genes (FDR<0.1 in both comparisons involving treatment)
```{r transitive, echo=FALSE, message=FALSE}
knitr::kable(
group_by(DE, Id, up) %>% summarise(n=n()) %>% filter(n>1) %>% group_by(up) %>% summarise(n=n())
)

knitr::kable(
group_by(DE, Id, down) %>% summarise(n=n()) %>% filter(n>1) %>% group_by(down) %>% summarise(n=n())
)


#group_by(DE, Id, up) %>% summarise(n=n()) %>% filter(n>1) %>% left_join(DE) %>% 
#  bind_rows(group_by(DE, Id, down) %>% summarise(n=n()) %>% filter(n>1) %>% left_join(DE)) %>%
#  select(Id, up, down, log2FoldChange, pvalue, padj, maxCooks) %>%
#  mutate(log2FoldChange = abs(log2FoldChange)) %>% filter(padj<1e-10) %>% View()


```

## Representative plots

```{r, echo=FALSE, message=FALSE}
filter(DE, Id=='6ADC146B50087543193CB22FCDAA4F44') %>% dplyr::slice(1) %>% plotExpression(LibraryInfo, "up-regulated in hog1 mutatant-infected")

filter(DE, Id=='AECA90DEE31652E43B963092F590FA95') %>% dplyr::slice(1) %>% plotExpression(LibraryInfo, "up-regulated in WT-infected")

filter(DE, Id=='3BA140C82516519D853ED9FA1670588B') %>% dplyr::slice(1) %>% plotExpression(LibraryInfo, "down-regulated in WT-infected; further down-regulated in hot1 mutant-infected")
filter(DE, Id=='37B8A1D6A6EED30CFB000CA7A056472D') %>% dplyr::slice(1) %>% plotExpression(LibraryInfo, "down-regulated in hog1 mutatant-infected")

filter(DE, Id=='815C82624BC0AC918E25D156CA177760') %>% dplyr::slice(1) %>% plotExpression(LibraryInfo, "up-regulated in WT-infected")

```